celery/backends/elasticsearch.py

```diff
@@ -88,6 +89,7 @@ def set(self, key, value):
             self._index(key, data, refresh=True)
 
     def _index(self, id, body, **kwargs):
+        body = dict((string(k), v) for k, v in body.items())
         return self.server.index(
```

Comment:
Totally unrelated but since you are using this backend perhaps you can help. Why isn't the `id` argument value passed to `index` here? Shouldn't this be:

```
return self.server.index(
    id=id,
    index=self.index,
    doc_type=self.doc_type,
    body=body,
    **kwargs
)
```